dsc/workflows/opencourseware.py

```diff
@@ -13,6 +14,234 @@
 logger = logging.getLogger(__name__)
 
 
+class OpenCourseWareTransformer:
+    """Transformer for OpenCourseWare (OCW) source metadata."""
+
+    fields: Iterable[str] = [
+        # fields with derived values
+        "dc_title",
+        "dc_date_issued",
+        "dc_description_abstract",
+        "dc_contributor_author",
+        "creativework_learningresourcetype",
+        "dc_subject",
+        "dc_identifier_other",
+        "dc_coverage_temporal",
+        "dc_audience_educationlevel",
+        # fields with static values
+        "dc_type",
+        "dc_rights",
+        "dc_rights_uri",
+        "dc_language_iso",
+    ]
+
+    @classmethod
+    def transform(cls, source_metadata: dict) -> dict:
+        """Transform source metadata."""
+        transformed_metadata = {}
+        for field in cls.fields:
+            field_method = getattr(cls, field)
+            formatted_field_name = field.replace("_", ".")
+
+            # check if 'source_metadata' is in signature
+            signature = inspect.signature(field_method)
+            if "source_metadata" in signature.parameters:
+                transformed_metadata[formatted_field_name] = field_method(source_metadata)
+            else:
+                transformed_metadata[formatted_field_name] = field_method()
+
+        return transformed_metadata
+
+    @classmethod
+    def dc_title(cls, source_metadata: dict) -> str:
+        """Build a title string from course numbers, title, and term year.
+
```

Comment:
I can't remember what we had before for `dc.title` but the complexity of this definitely justifies a dedicated method!